packages/react-form-hooks/src/hooks/field/field.ts

```diff
@@ -0,0 +1,196 @@
+ *
+ * @remarks
+ * **Reinitialization:** If the `value` property of the field configuration changes between calls to `useField`,
+ * the field will be reset to use it as it's new default value.
+ *
+ * **Imperative methods:** The returned `Field` object contains a number of methods used to imperatively alter it's state.
+ * These should only be used as escape hatches where the existing hooks and components do not make your life easy,
+ * or to build new abstractions in the same vein as `useForm`, `useSubmit` and friends.
+ */
+export function useField<Value = string>(
+  input: FieldConfig<Value> | Value,
+  dependencies: unknown[] = [],
+): Field<Value> {
+  const {value, validates} = normalizeFieldConfig(input);
+  const validators = normalizeValidation(validates);
+
+  const [state, dispatch] = useFieldReducer(value);
+
+  const reset = useCallback(() => dispatch(resetAction()), [dispatch]);
+  const newDefaultValue = useCallback(
+    value => dispatch(newDefaultAction(value)),
+    [dispatch],
+  );
+
+  const runValidation = useCallback(
+    (value: Value = state.value) => {
+      const error = validators
+        .map(check => check(value, {}))
+        .filter(value => value != null);
+
+      if (error && error.length > 0) {
+        const [firstError] = error;
+        dispatch(updateErrorAction(firstError));
+        return firstError;
+      }
+
+      dispatch(updateErrorAction(undefined));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state.value, ...dependencies],
+  );
+
+  const onChange = useCallback(
+    (value: Value | ChangeEvent) => {
+      const normalizedValue = isChangeEvent(value) ? value.target.value : value;
+
+      dispatch(updateAction(normalizedValue) as any);
+
+      if (state.error) {
+        runValidation(normalizedValue as Value);
+      }
+    },
+    [runValidation, state.error, dispatch],
+  );
+
+  const setError = useCallback(value => dispatch(updateErrorAction(value)), [
+    dispatch,
+  ]);
+
+  const onBlur = useCallback(
+    () => {
+      if (state.touched === false && state.error == null) {
+        return;
+      }
+
+      runValidation();
+    },
+    [runValidation, state.touched, state.error],
+  );
+
+  // We want to reset the field whenever a new `value` is passed in
+  useEffect(
+    () => {
+      newDefaultValue(value);
+    },
+    [value, newDefaultValue],
+  );
+
+  const field = useMemo(
+    () => {
+      return {
+        ...state,
+        onBlur,
+        onChange,
+        newDefaultValue,
+        runValidation,
+        setError,
+        reset,
+      };
+    },
+    [state, onBlur, onChange, newDefaultValue, runValidation, setError, reset],
+  );
+
+  return field as Field<Value>;
+}
+
+function normalizeFieldConfig<Value>(
+  input: FieldConfig<Value> | Value,
+): FieldConfig<Value> {
+  if (input == null || typeof input !== 'object') {
+    return {value: input, validates: () => undefined};
```

Comment:
Weird that `validates` isn't just optional?

Reply:
we could simplify this stuff if we made the signature always
```
(config: {value: Value, validates?: Validator<Value>}, dependencies?: unknown[]): Field<Value>;
```
but it would mean that for trivial fields you'd need to do `useField({value: 'foo'});` which is a bit annoying